### PR TITLE
feat(alerting): Implement Zulip's alerts

### LINKED ISSUE #802: Alert send to Zulip
### Describe the feature request

Receive notifications and if possible a status picture 

### Why do you personally want this feature to be implemented?

We use the internal messenger zulip and it would be handy to get notifications into it

### How long have you been using this project?

I'm just getting acquainted so I can put it into practice

### Additional information

docs https://zulip.com/api/incoming-webhooks-overview

### PR DESCRIPTION
**Pull request by mehdibo**
## Summary
Fixes #802

## Checklist
- [x] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [X] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

**Comment by mehdibo** [comment]
Here is a preview of the message sent (the last one is the one implemented):
https://chat.zulip.org/#narrow/stream/7-test-here/topic/Gatus

File: alerting/config.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package alerting
 
 import (
+	"github.com/TwiN/gatus/v5/alerting/provider/zulip"
 	"log"
 	"reflect"
 	"strings"
```
Comment (TwiN) [inline alerting/config.go:7]: Can you move the `"github.com/TwiN/gatus/v5/alerting/provider/zulip"` import to the 2nd section with the other non-standard imports?

File: alerting/provider/zulip/zulip.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import (
+	"bytes"
+	"fmt"
+	"github.com/TwiN/gatus/v5/alerting/alert"
+	"github.com/TwiN/gatus/v5/client"
+	"github.com/TwiN/gatus/v5/config/endpoint"
+	"io"
+	"net/http"
+	"net/url"
+)
```
Comment (TwiN) [inline alerting/provider/zulip/zulip.go:12]:
```suggestion
import (
	"bytes"
	"fmt"
	"io"
	"net/http"
	"net/url"

	"github.com/TwiN/gatus/v5/alerting/alert"
	"github.com/TwiN/gatus/v5/client"
	"github.com/TwiN/gatus/v5/config/endpoint"
)
```

File: alerting/provider/zulip/zulip.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+type Config struct {
+	// BotEmail is the email of the bot user
+	BotEmail string `yaml:"bot-email"`
+	// BotAPIKey is the API key of the bot user
+	BotAPIKey string `yaml:"bot-api-key"`
+	// Domain is the domain of the Zulip server
+	Domain string `yaml:"domain"`
+	// ChannelId is the ID of the channel to send the message to
+	ChannelId string `yaml:"channel-id"`
```
Comment (TwiN) [inline alerting/provider/zulip/zulip.go:22]: Can you rename `ChannelID` to `ChannelId`?

**Review by TwiN — CHANGES_REQUESTED** [review]
Great work so far!

Could you also add the new provider here: https://github.com/TwiN/gatus/blob/7de5a1fe48655d5ba732118029354ea522f5d79f/alerting/provider/provider.go#L62-L84

File: alerting/provider/zulip/zulip.go (hunk 3 above)
Comment (mehdibo) [reply alerting/provider/zulip/zulip.go:23]: You mean `ChannelId` to ChannelID` right?

Comment (mehdibo) [reply alerting/provider/zulip/zulip.go]: @TwiN I assumed `ChannelId` to `ChannelID` and pushed it, let me know if there is anything else

File: alerting/provider/zulip/zulip_test.go [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+
+import (
+	"fmt"
+	"github.com/TwiN/gatus/v5/alerting/alert"
+	"github.com/TwiN/gatus/v5/client"
+	"github.com/TwiN/gatus/v5/config/endpoint"
+	"github.com/TwiN/gatus/v5/test"
+	"net/http"
+	"net/url"
+	"testing"
```
Comment (TwiN) [inline alerting/provider/zulip/zulip_test.go:11]:
```suggestion
	"fmt"
	"net/http"
	"net/url"
	"testing"

	"github.com/TwiN/gatus/v5/alerting/alert"
	"github.com/TwiN/gatus/v5/client"
	"github.com/TwiN/gatus/v5/config/endpoint"
	"github.com/TwiN/gatus/v5/test"
```

**Comment by TwiN** [comment]
@mehdibo Great work!

**Review by TwiN — APPROVED** [review]

**Pull request merged**